File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
class Solution {
    public String solution(int n) {
        
        String word[] = {"4", "1", "2"};
        String answer = "";
        
        while(n > 0){
            int mod = n%3;
            answer = word[mod]+answer;
            n /= 3;
            if(mod==0) n--;
        }
        
        return answer;                                                                
    }
}

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.js
function solution(n) {
    
    const word = ["4", "1", "2"]
    var answer = '';
    
    while(n > 0){
        let mod = n%3;
        answer = word[mod] + answer;
        n = Math.floor(n/3);
        if(mod == 0) n--;
    }
    
    return answer;
}